src/wakelock.c

```diff
@@ -61,7 +62,7 @@ void belle_sip_wake_lock_uninit(JNIEnv *env) {
  * @param data Unused.
  */
 static void jni_key_cleanup(void *data) {
-	JNIEnv *env = pthread_getspecific(ctx.jniEnvKey);
+	JNIEnv *env = (JNIEnv*) data;
```

Comment:
http://linux.die.net/man/3/pthread_key_create

In the documentation of `pthread_key_create`, it says that upon thread destruction, the value is set to NULL and *then* the destructor (this function) is called with the value as an argument.  

> An optional destructor function may be associated with each key value. At thread exit, if a key value has a non-NULL destructor pointer, and the thread has a non-NULL value associated with that key, the value of the key is set to NULL, and then the function pointed to is called with the previously associated value as its sole argument. The order of destructor calls is unspecified if more than one destructor exists for a thread when it exits. 

When I read this, I don't understand how this ever worked.  It looks like `pthread_getspecific()` is supposed to return NULL when called from the destructor (this function).